docs/guides/blocks/sync-tables/two-way.md

```diff
@@ -0,0 +1,300 @@
+    }
+  },
+});
+```
+
+When processing updates in batches be sure to return the same number of results as there were updates, in the same order. If an individual update fails, catch the exception and return it in the results array to indicate that the row update failed. Unhandled exceptions will prevent the entire batch from completing and won't give the user a clear indication of which row was the cause. See the [Handling errors](#handling-errors) section below for more information.
+
+
+## Property autocomplete {: #autocomplete}
+
+Similar to [parameter autocomplete][parameters_autocomplete], you can provide a suggested set of values for mutable properties, which will be shown to the user when they are editing a corresponding cell.
+
+<img src="../../../../images/two_way_autocomplete.png" srcset="../../../../images/two_way_autocomplete_2x.png 2x" class="screenshot" alt="Autocomplete in a cell.">
+
+The valid options are defined in the `autocomplete` field of the property, which can be either an array of static values or a function that generates them dynamically. An autocomplete function can access the value of other properties in the row via `context.propertyValues` and the current search string typed by the user via `context.search`.
+
+```ts
+const ShirtSchema = coda.makeObjectSchema({
+  properties: {
+    size: {
+      type: coda.ValueType.String,
+      autocomplete: ["S", "M", "L", "XL"],
+    },
+    color: {
+      type: coda.ValueType.String,
+      autocomplete: async function (context) {
+        let size = context.propertyValues.size;
+        let availableColors = await getColorsForSize(context, size);
+        return availableColors;
+      },
+    },
+    pattern: {
+      type: coda.ValueType.String,
+      autocomplete: async function (context) {
+        let search = context.search;
+        let patterns = await searchPatterns(context, search);
+        return patterns;
+      },
+    },
+  },
+  // ...
+});
+```
+
+For sync tables with dynamic schemas, you aren't allowed to define the search function directly on the property itself. Instead use the special value `AutocompleteType.Dynamic`, which tells Coda to call the sync table's `autocomplete` function. This is a single function that handles the auto-complete for all dynamic properties. It's defined directly on a dynamic sync table, or within the `dynamicOptions` of a regular sync table. The function can determine which property to provide autocomplete for by inspecting `context.propertyName`.
+
+=== "Dynamic sync table"
+
+    ```ts
+    pack.addDynamicSyncTable({
+      // ...
+      getSchema: async function (context) {
+        let attributes = await getCustomAttributes(context);
+        for (let attr of attriutes) {
+          properties[attr] = {
+            // ...
+            autocomplete: coda.AutocompleteType.Dynamic,
+          }
+        }
+        // ...
+      },
+      autocomplete: async function (context) {
+        let attr = context.propertyName;
```

Comment:
Now that we're writing documentation, we'll need to make a decision about whether `propertyName` is the normalized property name, the `fromKey`, or the key originally written by the pack maker, along with the related question about whether to denormalize schemas in the formula context also

We'll probably need to schedule some time to continue discussing and see if we can agree on a path forward.

Reply:
Ya, I've added a TODO to go back and document this once we have it finalized.